### PR TITLE
kube-aws: validate externalDNSName is a subdomain of hostedZone

### LINKED ISSUE #401: externalDNSName is not validated against hostedZone
```yaml
createRecordSet: true
hostedZone: staging.core-os.net
externalDNSName: my-cluster.staging2.core-os.net
```

`kube-aws render` will still succeed, even though we know that this cluster config is invalid because `externalDNSName` is not a subdomain of `hostedZone`.

@cgag can you add:
* a validation check in `Config.valid()`, which checks that `externalDNSName` is a valid sub-domain of `hostedZone`?
* tests in `config_test.go` which test valid and invalid relationship between the two fields?

### PR DESCRIPTION
**Pull request by cgag**
fixes #401

### CODE REVIEW

**Comment by colhom** [comment]
@cgag can you rebase?

File: multi-node/aws/pkg/config/config_test.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ import (
 	"testing"
 )
 
-const minimalConfigYaml = `externalDNSName: test-external-dns-name
+const minimalConfigYaml = `externalDNSName: test.example.com
```
Comment (colhom) [inline multi-node/aws/pkg/config/config_test.go]: should we make this `staging.core-os.net` to be consistent with `cluster_test.go`?

**Comment by colhom** [comment]
I would throw another test in there which tests `externalDNSName=a.b.c.staging.core-os.net` and `hostedZone=staging.core-os.net`

**Comment by cgag** [comment]
Added that test, made the minimalConfigYaml's consistent, rebased.

File: multi-node/aws/pkg/config/config_test.go [hunk 2]
```diff
@@ -102,6 +105,10 @@ hostedZone: ""
 # recordSetTTL shouldn't be modified when createRecordSet is false
 createRecordSet: false
 recordSetTTL: 400
+`, `
+# whatever.com is not a superdomain of test.example.com
```
Comment (colhom) [inline multi-node/aws/pkg/config/config_test.go]: "...not a superdomain of test.staging.core-os.net"

**Comment by cgag** [comment]
Fixed the outdated "whatever is not a superdomain..."

**Comment by colhom** [comment]
LGTM. Thanks @cgag

**Pull request merged**